src/test/java/core/basesyntax/service/impl/ReaderServiceImplTest.java

```diff
@@ -0,0 +1,69 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReaderService;
+import core.basesyntax.service.TransactionException;
+import java.io.File;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ReaderServiceImplTest {
+    private static final String EXCEPTION_MESSAGE
+            = TransactionException.class.toString();
+    private static final String EMPTY_CSV = "src/test/resources/empty.csv";
+    private static final String INCORRECT_FILE_PATH_FORMAT = "src/test/resources/input.mp3";
+    private static final String INPUT_CSV_WITH_CONTENT = "src/test/resources/input.csv";
+    private static ReaderService readerService;
+
+    @BeforeAll
+    static void beforeAll() {
+        readerService = new ReaderServiceImpl();
+    }
+
+    @Test
+    void reader_nullFilePath_notOk() {
+        Assertions.assertThrows(TransactionException.class,
+                () -> readerService.readFromFile(null),
+                String.format("%s should throw when filePath is null", EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void reader_validData_Ok() {
+        Assertions.assertNotNull(readerService.readFromFile(INPUT_CSV_WITH_CONTENT));
+    }
+
+    @Test
+    void reader_emptyData_notOk() {
+        Assertions.assertThrows(TransactionException.class,
+                () -> readerService.readFromFile(EMPTY_CSV),
+                String.format("%s should throw when input file is empty", EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void reader_validFileFormat_notOk() {
+        Assertions.assertThrows(TransactionException.class,
+                () -> readerService.readFromFile(INCORRECT_FILE_PATH_FORMAT),
+                String.format("%s should throw when file have incorrect format",
+                        EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void reader_validFilePath_Ok() {
+        readerService.readFromFile(INPUT_CSV_WITH_CONTENT);
+    }
+
+    @Test
+    void reader_fileCreatedSuccessfully_Ok() {
+        readerService.readFromFile(INPUT_CSV_WITH_CONTENT);
+        File file = new File(INPUT_CSV_WITH_CONTENT);
+        Assertions.assertTrue(file.exists());
```

Comment:
why are you testing that file exists in ReaderService? Even if your test file didn't exist, it would only mean that you have written the test wrong, not that your service doesn't work :)

Reply:
Yeah, totally agree. To my mind when I wrote that test, I think how to do a lot as tests as I can. So, as a result, I added code for creating file in class which should just read from file :(